### ISSUE TITLE
No match found for mapping?

### ISSUE THREAD

**Issue by Silverwing9** [Norbyte/bg3se#66]
I get 4 weird errors when starting BG3 post-patch 3 in the console, and I'm not sure what they're referring to and how to fix it, and there's also something about a missing spell component, not sure if thats mod related or console
![e457b5ee32cd1a17f968a14da34cd5ea](https://github.com/Norbyte/bg3se/assets/143288712/9e7bba8a-ddc0-437e-a381-ce6fe5e253cb)

**Comment by Pyr0blad3** [comment]
thanks, right now i wanted to post the exact same thing as an issue here.
kinda funny but yeah i get the same messages when starting the game (patch 3 installed), no matter if i have mods active or not.

maybe a side note: loading save files, or creating a new game wont throw any more "error" messages. at least for me thats the case.

hope its just a error that doesnt matter. cheers and thanks for your work and help!

![2023-09-23 00_41_29-Window](https://github.com/Norbyte/bg3se/assets/28388430/17329ba8-230d-4ef7-a12e-48a7c56407ec)

**Comment by Bnidoya** [comment]
My console also displays like this, my game has always been CTD since the update, which makes me very sad

**Comment by Pyr0blad3** [comment]
@Bnidoya are you not able to play the game even with this messages in the console? 
i can load saves and create new games, even with those messages.

**Comment by Silverwing9** [comment]
> thanks, right now i wanted to post the exact same thing as an issue here. kinda funny but yeah i get the same messages when starting the game (patch 3 installed), no matter if i have mods active or not.
> 
> maybe a side note: loading save files, or creating a new game wont throw any more "error" messages. at least for me thats the case.
> 
> hope its just a error that doesnt matter. cheers and thanks for your work and help!
> 
> ![2023-09-23 00_41_29-Window](https://user-images.githubusercontent.com/28388430/270070018-17329ba8-230d-4ef7-a12e-48a7c56407ec.png)
> 
> edit: @Silverwing9 even the BG3Ext v5 built on Sep "Time" was the same lol!

Haha thats such a funny coincidence
I can load saves and make new games, but what I can't do is confirm my choices with the magic mirror. I can get in and change things but I can't confirm and get out

**Comment by Pyr0blad3** [comment]
@Silverwing9 @Bnidoya 

i found this: https://github.com/Norbyte/bg3se/issues/65 (already closed issue)
where @Norbyte is saying "The errors mentioned above are 'normal' and will always appear."

so i think those errors doesnt matter at all for your game, BUT it still would be nice to get a second confirmation on this i guess.

**Comment by Bnidoya** [comment]
> @Silverwing9 @Bnidoya
> 
> 我发现这个：#65（已经关闭的问题）在哪里说“上面提到的错误是'正常'的，并且会一直出现。
> 
> 所以我认为这些错误对你的游戏来说根本不重要，但我想得到第二次确认仍然会很好。

oh ty my man!!

**Comment by MaravicMarko** [comment]
Here is my contribution:

![image](https://github.com/Norbyte/bg3se/assets/72248348/96e607b4-7150-4339-b35a-2f71206c93a1)

Started after latest update of BG3 (v4.1.1.3732833), game just crashes after GameWorks/Larian video. 
Weird thing is that on screenshot of SE is written "BG3Ext v2", but I have installed v3 :-/
And just to note, before I didn't have any error messages in SE console.

**Comment by Pyr0blad3** [comment]
> Started after latest update of BG3 (v4.1.1.3732833), game just crashes after GameWorks/Larian video. Weird thing is that on screenshot of SE is written "BG3Ext v2", but I have installed v3 :-/ And just to note, before I didn't have any error messages in SE console.


i am not 100% sure about this, but i have read in other issues that ScriptExtender should update itself automatically through BG3 Mod Manager i think.

at least there is no "new file" to download or anything but i think my ScriptExtender updated today.

ALSO idk if you uploaded the wrong screenshots but your console logs are from 31.08.2023... which is serveral weeks in the past. "BG3Ext v2" writte in your console and we have "BG3Ext v5" written in ours, which is another indicator for why it either isnt working or that you uploaded an old/wrong console log screenshot.

please if you have the same problem upload the write screenshots or change your comment so nobody gets confused.
if i am mistaken i am sorry but i dont think i am.
cheers.

**Comment by MaravicMarko** [comment]
> > Started after latest update of BG3 (v4.1.1.3732833), game just crashes after GameWorks/Larian video. Weird thing is that on screenshot of SE is written "BG3Ext v2", but I have installed v3 :-/ And just to note, before I didn't have any error messages in SE console.
> 
> @MaravicMarko i am not 100% sure about this, but i have read in other issues that ScriptExtender should update itself automatically through BG3 Mod Manager i think.
> 
> at least there is no "new file" to download or anything but i think my ScriptExtender updated today.
> 
> ALSO idk if you uploaded the wrong screenshots but your console logs are from 31.08.2023... which is serveral weeks in the past. "BG3Ext v2" writte in your console and we have "BG3Ext v5" written in ours, which is another indicator for why it either isnt working or that you uploaded an old/wrong console log screenshot.
> 
> please if you have the same problem/question about this topic as we do, upload the right screenshots or change your comment so nobody gets confused with either 2 seperat problems, or only 1 problem but with wrong console logs as comments. if i am mistaken i am sorry but i dont think i am. cheers.
> 
> edit: i would try reinstalling the ScriptExtender File (latest available version) first, if nothing works. https://github.com/Norbyte/bg3se/releases - (dont download from ASSETS!)

@Pyr0blad3 Hi, thanks for quick response. 

I understand that screenshots are contradictory, but they were created 5 minutes apart.
What is confusing for me is why "Extender DLL" is being loaded from "AppData/Local/BG3ScriptExtender" location (as seen on second screenshot on my prior post).

Here are screenshots of content of said folder:

![image](https://github.com/Norbyte/bg3se/assets/72248348/f4a80c93-6605-4fb5-b919-ff655fb8c3a7)
![image](https://github.com/Norbyte/bg3se/assets/72248348/b63f429b-9ea6-48c0-84ff-6ba57c0638ff)
![image](https://github.com/Norbyte/bg3se/assets/72248348/d40a99fb-4f37-4e94-aa4c-60d5f5199f7b)
![image](https://github.com/Norbyte/bg3se/assets/72248348/f2529f31-3a8e-479a-9e51-b8f532a62217)

To be honest, I don't remember extracting/installing SE to this location.

I have also tried to update SE through BG3 Mod Manager (latest version from GitHub):

![image](https://github.com/Norbyte/bg3se/assets/72248348/55bcc9fe-ef4a-4f16-9b7b-81a8dc71326d)

Also, for some reason, SE is reported as v1 in BG3MM (if I understood tooltip correctly):
 
![image](https://github.com/Norbyte/bg3se/assets/72248348/f57e7748-60cd-49da-862f-65363c1a46b9)

And I have also tried with the latest version from GitHub, and manually extracted "DWrite.dll" to game "bin" folder overwriting existing file.

If I can do some more "diagnostic activities", please let me know.

Cheers!

**Comment by Pyr0blad3** [comment]
@MaravicMarko 

**First of all, sorry for the accusation, but as you also said the time was a bit contradicting and I would guess you have for now a different problem than the others here do.**
So basically i have no clue how ScriptExtender works or anything like that but at least i have programming/coding knowlledge, so I looked though my Folders and Files too.

**Second of all to make things clear where i can: the v1 in BG3 Mod Manager is normal (I have it also even tho I use v5 of ScriptExtender in the Console Log). So there is no worry there at least.**

Now after reading your comment I looked at "AppData/Local/BG3ScriptExtender" and I think it will make sence why stuff is loaded from there in just a moment I think.

**TALKING ABOUT MY FOLDER STUCTURE AND FILES NOW:**
The first folder (BG3ScriptExtender) looks like yours in the first screenshot you did.
The second folder (BG3ScriptExtender/ScriptExtender) looks way different to yours in the second screenshot you did.

**//Folder: ScriptExtender - Entries**
![2023-09-23 02_38_39-Window](https://github.com/Norbyte/bg3se/assets/28388430/621a0f2d-5ee2-4e1c-a0f3-301372a30c1e)

I have highlighted the v5 content for ScriptExtender I have installed, which you are missing.
**This would be the problem, as you cant load a "newer version" (v5 in this case) when you dont have the files.**
Now the question why the files for this wont get downloaded automatically is a mistery for me too, BUT it seems as you have run an old version of ScriptExtender for a longer time, because you dont even have v3 or v4 listed in this folder (BG3ScriptExtender/ScriptExtender)



**Going on I went back to the last folder (BG3ScriptExtender) and looked at the "Manifest-Release" file:**

**//File in Notpad++**
![2023-09-23 02_39_11-Window](https://github.com/Norbyte/bg3se/assets/28388430/a1f8c602-c375-4266-87c3-84a7bbe8de0a)

Again I have highlighted the most imporant part you are missing, which is the integration of v5 (newest version of ScripExtender)
Somewhere when starting ScriptExtender (when you launch BG3 through the Mod Manager) the ScriptExtender itself probably looks through those files and folders to get the "newest version" and config loaded.

No problem if it doesnt make sence to you but there is 100% something that is not correct in your files (if you want the newest version).
**Sadly I dont know how to fix it for you, again I would try uninstalling all ScriptExtender files (so steam/.../Bg3/bin/"ScriptExtenderFile") and the folder we just talked about -> launching the game -> closing it -> reinstalling a new ScriptExtender file -> launch the game again -> check BG3ScriptExtender folder if something changed.**

in case that doesnt work I wil lpost my files/code here so you can try copy it. that could also work but again as i dont have a clue how the ScriptExtender works I sadly cant fix it just like that. ALSO i am not the maker of ScriptExtender so take everything I am saying with a grant of salt i guess.

I hope I could help a little bit.
cheers and good luck!


Download Files/Folder v5:
[v5.zip](https://github.com/Norbyte/bg3se/files/12705262/v5.zip)

**Comment by Norbyte** [comment]
Relevant portion of the log:
![image](https://github.com/Norbyte/bg3se/assets/2243424/0bc92c42-1d9d-4520-bcf3-ccfd48ef2082)

Your firewall is blocking connection to the update server.

**Comment by Pyr0blad3** [comment]
@Norbyte as you didnt mention them, 
I would assume that the 4 "no match found mapping" errors I and other people get in ScriptExtender Console on StartUp, 
can be ignored?

**Comment by MaravicMarko** [comment]
> Relevant portion of the log: ![image](https://user-images.githubusercontent.com/2243424/270098914-0bc92c42-1d9d-4520-bcf3-ccfd48ef2082.png)
> 
> Your firewall is blocking connection to the update server.

@Pyr0blad3 thanks for extensive feedback. I have used v5 from link you shared and now everything works.

@Norbyte Thanks for reply. I realized that, but I don't have anything on block list. I even created allow rule for all BG3 exe files. How to unblock SE?
I manually "instaled" v5, but I am still getting connection error message when game loads. Aside from that, everything is OK now, no more CTDs.

In any case, thanks for your work and effort on SE 🖖

**Comment by belowzeros** [comment]
going through the same thing myself, the firewall thing I don't know if that's the answer.  I turned mine off and still get the can't connect problem.

**Comment by sheldoncooper8668** [comment]
Mine was updated well to v5, but a crash occurred.
* 100% crash when talking to Netty in Druid forest.

It works normally after replacing it with the v5 file you attached.

Please consider distributing new files instead of automatic updates.

**Comment by Pyr0blad3** [comment]
@sheldoncooper8668 you mean my file?
maybe you had an "older" version of v5? mine updated 2 times now i think.

BUT yeah i hear a lot of "cant autoUpdate ScriptExtender" right now from people. I honestly dont know why its working for me. do you use normal windows firewall or an external because I use an 3rd party firewall i paid for. maybe its some problem with windows firewall? idk.

@belowzeros @sheldoncooper8668 @MaravicMarko 
if this issue is still present in v6 or higher (whenever ScriptExtender goes up a version) I will try to provide my file in some post here on github again.

so you might wanna remember that just in case you encounter that problem again.

**Comment by Valky93** [comment]
I got the same problem but i think i got where is the issue. 

I have no problem loading the game or even saving but as soon as i open a shop with some NPC the next time i try to save i have the error code 516 

Maybe those precision can help fix it ? 

Thx a lot btw

**Comment by Pyr0blad3** [comment]
v6 is here: https://github.com/Norbyte/bg3se/issues/90#issuecomment-1746715480 just in case.

**Comment by DarBear808** [comment]
![image_2023-11-02_035829501](https://github.com/Norbyte/bg3se/assets/122306815/e948d270-e696-4202-8942-ef3b103b63ec)

I recently played Baldur's Gare 3 and haven't played it in a week so I was surprised about these errors when I launch the game on the mod manager it runs fine but there are some issues. One mod that I noticed that didn't work was "Teleport Party To You Spell" mod. Another issues is that I do not have icons for items in containers, I would have to hover over the empty boxes with my mouse to see if there is a item inside the containers. I do not know if these errors are related to my problem or can I just ignore them

**Comment by DarBear808** [comment]
> ![image_2023-11-02_035829501](https://user-images.githubusercontent.com/122306815/280012571-e948d270-e696-4202-8942-ef3b103b63ec.png)
> 
> I recently played Baldur's Gare 3 and haven't played it in a week so I was surprised about these errors when I launch the game on the mod manager it runs fine but there are some issues. One mod that I noticed that didn't work was "Teleport Party To You Spell" mod. Another issues is that I do not have icons for items in containers, I would have to hover over the empty boxes with my mouse to see if there is a item inside the containers. I do not know if these errors are related to my problem or can I just ignore them

I think I realize the problem, there was a huge update recently and that could be the reason for some of my mods not working

**Comment by Norbyte** [comment]
I'm closing old issues related to old game versions, since those were largely fixed by later game patches and SE updates and are no longer relevant/reproducible.
